lib/macho/macho_file.rb

```diff
@@ -233,10 +264,12 @@ def dylib_id=(new_id)
       raise ArgumentError.new("new ID must be a String") unless new_id.is_a?(String)
       return unless dylib?
 
-      dylib_cmd = command(:LC_ID_DYLIB).first
-      old_id = dylib_id
+      old_cmd = command(:LC_ID_DYLIB).first
+      new_cmd = LoadCommand.create(:LC_ID_DYLIB, new_id, old_cmd.timestamp,
+        old_cmd.current_version, old_cmd.compatibility_version)
 
-      set_name_in_dylib(dylib_cmd, old_id, new_id)
+      add_command(new_cmd)
+      delete_command(old_cmd)
```

Comment:
Yes, currently we do a straight replacement of the zero bytes in `add_command`. But while it is a bit less efficient, that's effectively the same as inserting the serialized new command at the end of the current command list, then recalculating the total size of the new command list, and then de-padding at the end of that list. There's really no difference from an ordinary insert that operates in the middle of the command list, except for the offset.

**Deletion:**

```
aaabbcccc00000   # (0) original
aaacccc00000     # (1) removed load command b
aaacccc0000000   # (2) inserted zero bytes after last load command (pad)
```

**Insertion:**

```
aaabbcccc00000   # (0) original
aaaddbbcccc00000 # (1) inserted load command d before b
aaaddbbcccc000   # (2) removed zero bytes after last load command (de-pad)
```

**Addition (insertion at the end):**

```
aaabbcccc00000   # (0) original
aaabbccccdd00000 # (1) inserted load command d after c (before trailing zero bytes)
aaabbccccdd000   # (3) removed zero bytes after last load command (de-pad)
```

Does this make sense? If you still see a problem, which of the steps is problematic?